agent/api/ecsclient/client.go

```diff
@@ -312,12 +312,6 @@ func (client *APIECSClient) SubmitTaskStateChange(change api.TaskStateChange) er
 		return nil
 	}
 
-	// Submit task state change
-	if change.Status == api.TaskStatusNone {
-		seelog.Warnf("SubmitTaskStateChange called with an invalid change: %s", change.String())
-		return errors.New("ecs api client: SubmitTaskStateChange called with an invalid change")
-	}
-
 	if change.Status != api.TaskRunning && change.Status != api.TaskStopped && len(change.Containers) == 0 {
```

Comment:
let's also get rid of this block altogether. This determination should be happening in `sendableEvent .taskShouldBeSent` method: https://github.com/richardpen/amazon-ecs-agent/blob/fix-1075/agent/eventhandler/task_handler_types.go#L67